cdoc/job-tutorial.md

```diff
@@ -0,0 +1,268 @@
+    }
+  ],
+  "gpuType": String,
+  "killAllOnCompletedTaskNumber": Integer,
+  "retryCount": Integer
+}
+```
+
+下表为配置文件中每个参数的详细说明：
+| Field Name                     | Schema                     | Description                              |
+| :----------------------------- | :------------------------- | :--------------------------------------- |
+| `jobName`                      | String in `^[A-Za-z0-9\-._~]+$` format, required | Name for the job, need to be unique |
+| `image`                        | String, required           | URL pointing to the Docker image for all tasks in the job |
+| `authFile`                     | String, optional, HDFS URI | Docker registry authentication file existing on HDFS |
+| `dataDir`                      | String, optional, HDFS URI | Data directory existing on HDFS          |
+| `outputDir`                    | String, optional, HDFS URI | Output directory on HDFS, `$PAI_DEFAULT_FS_URI/Output/$jobName` will be used if not specified |
+| `codeDir`                      | String, optional, HDFS URI | Code directory existing on HDFS          |
+| `virtualCluster`               | String, optional           | The virtual cluster job runs on. If omitted, the job will run on `default` virtual cluster    |
+| `taskRoles`                    | List, required             | List of `taskRole`, one task role at least |
+| `taskRole.name`                | String in `^[A-Za-z0-9._~]+$` format, required | Name for the task role, need to be unique with other roles |
+| `taskRole.taskNumber`          | Integer, required          | Number of tasks for the task role, no less than 1 |
+| `taskRole.cpuNumber`           | Integer, required          | CPU number for one task in the task role, no less than 1 |
+| `taskRole.memoryMB`            | Integer, required          | Memory for one task in the task role, no less than 100 |
+| `taskRole.gpuNumber`           | Integer, required          | GPU number for one task in the task role, no less than 0 |
+| `taskRole.portList`            | List, optional             | List of `portType` to use                |
+| `taskRole.portType.label`      | String in `^[A-Za-z0-9._~]+$` format, required | Label name for the port type |
+| `taskRole.portType.beginAt`    | Integer, required          | The port to begin with in the port type, 0 for random selection |
+| `taskRole.portType.portNumber` | Integer, required          | Number of ports for the specific type    |
+| `taskRole.command`             | String, required           | Executable command for tasks in the task role, can not be empty |
+| `gpuType`                      | String, optional           | Specify the GPU type to be used in the tasks. If omitted, the job will run on any gpu type |
+| `killAllOnCompletedTaskNumber` | Integer, optional          | Number of completed tasks to kill the entire job, no less than 0 |
+| `retryCount`                   | Integer, optional          | Job retry count, no less than 0          |
```

Comment:
the table's format has been changed, please view it at https://github.com/YitongFeng/pai/blob/master/cdoc/job-tutorial.md

Reply:
OK, thanks.